partition_table.go

```diff
@@ -137,27 +138,34 @@ func (p *PartitionTable) loadRestarting(ctx context.Context, stopAfterCatchup bo
 	var (
 		resetTimer *time.Timer
 		retries    int
+		retryMux   sync.Mutex
```

Comment:
instead of the mutex, we could use `atomic.Int32`, do you think that would improve readability?